src/ziggurat/streams.clj

```diff
@@ -39,32 +43,52 @@
   (if (some? upgrade-from-config)
     (.put properties StreamsConfig/UPGRADE_FROM_CONFIG upgrade-from-config)))
 
+(defn- set-encoding-config
+  "Populates `key-deserializer-encoding`, `value-deserializer-encoding` and `deserializer-encoding`
+   in `properties` only if non-nil."
+  [properties key-deserializer-encoding value-deserializer-encoding deserializer-encoding]
+  (if (some? key-deserializer-encoding)
+    (.put properties KEY_DESERIALIZER_ENCODING key-deserializer-encoding))
+  (if (some? value-deserializer-encoding)
+    (.put properties VALUE_DESERIALIZER_ENCODING value-deserializer-encoding))
+  (if (some? deserializer-encoding)
+    (.put properties DESERIALIZER_ENCODING deserializer-encoding)))
```

Comment:
what are the default values for these?

Reply:
is this config only valid when the serializer is a string serializer?

Reply:
Default value is "UTF8" which is hard-coded in both StringSerializer and StringDeserializer. Hence, we don't need to provide our own. Please see [1], for example.

Yes, only StringSerializer uses this configuration.

[1] https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/serialization/StringDeserializer.java